include/pybind11/pytypes.h

```diff
@@ -491,7 +491,7 @@ class tuple : public object {
         if (!m_ptr) pybind11_fail("Could not allocate tuple object!");
     }
     size_t size() const { return (size_t) PyTuple_Size(m_ptr); }
-    detail::tuple_accessor operator[](size_t index) const { return detail::tuple_accessor(ptr(), index); }
```

Comment:
No problem.

An additional note (as a record): [the relevant changeset](https://github.com/yungyuc/pybind11/commit/48548ea4a5735acfb86c67354d3ea6a728169740#diff-0e12743ab8ce3cfb72c5df5d9b956b47R183) shows that both `tuple::tuple()` and `list::list()` changed to take and store `handle`, but only `list::operator[]()` changed to feed `*this`.